api/comment_threads.rb

```diff
@@ -66,7 +66,21 @@
 
 put "#{APIPREFIX}/threads/:thread_id" do |thread_id|
   filter_blocked_content params["body"]
-  thread.update_attributes(params.slice(*%w[title body pinned closed commentable_id group_id thread_type]))
+  updated_content = params.slice(*%w[title body pinned closed commentable_id group_id thread_type edit_reason_code close_reason_code])
+  # Only set edited_by if the content has been edited by another user
+  if updated_content.has_key? BODY and thread.author != user
+    updated_content["edited_by"] = user
+    updated_content["previous_body"] = thread.body
```

Comment:
This will overwrite the previous body every time an edit is made, so if it's edited twice, the original body is lost.
Wouldn't it be better if the original post was kept?

Reply:
@felipetrz  The full moderation history will be captured on the edx-platform side: https://openedx.atlassian.net/browse/TNL-8019

As for whether this should retain the original vs previous body, I think you have a good point. My thinking here was that retaining the previous body would allow seeing what changed in each edit. 

I think keeping the original body makes more sense, so will modify this. In any case, edx-platform will keep a history of all edits.

Reply:
@felipetrz  I've decided to keep the moderation history here and updated the PR accordingly. Now each time you make an edit, it will keep the previous body, and the reason code for the edit along with the user and timestamp.